### ISSUE TITLE
modesetting xorg driver

### ISSUE THREAD

**Issue by jsorg71** [01org/libyami#862]
Of course yami works great with the intel xorg driver when configured with --enable-x11.
I noticed that newer intel system default to modesetting driver which uses glamor to accelerate the xserver with opengl.
yami should work with modesetting driver as well when --enable-x11 us used?
Do you guys use intel or modesetting driver?

**Comment by xuguangxin** [comment]
We use Xorg driver. Could you share more details about the glamor and modeseting driver?

yami works fine with dma_buf. Basically, if you can get dma_buf handle from your display.   
We have [grid](https://github.com/intel/libyami-utils/blob/master/examples/grid.cpp) example to show you how to use libdrm to display directly.  We also have [example ](https://github.com/intel/libyami-utils/blob/master/egl/egl_vaapi_image.h)to show you how to share buffers with Opengl .
Not sure it can address your problem or not.

thanks

**Comment by jsorg71** [comment]
If you go out and buy a new i7 processor and install Debian 9 or Ubuntu 18.04, you will not get the Intel Xorg driver by default.  You will get the modesetting driver.  With this, va_x11.h functions seem to work but I see corrupted frame once in a while.  Still trying to determine what going on.  I think this is not a yami issue but just want to bring it up.
BTW, thanks for the example links.

**Comment by jsorg71** [comment]
I have a bit more info on this.  First note this Debian announcement
https://tjaalton.wordpress.com/2016/07/23/intel-graphics-gen4-and-newer-now-defaults-to-modesetting-driver-on-x/
Also, I stated i7 but even low end Atom processor NUC like 
https://ark.intel.com/products/78577/Intel-NUC-Kit-DE3815TYKHE
will default modesetting Xorg driver.
I tried displaying with GLX, same problem as X11.  I was able to get it working with DRI3 with some hacking.
This is certainly not a yami bug and not even a libva.  I'm pretty sure this is a glamor issue with a missing glFlush or glFinish.  I'll close this and find the modesetting bug tracker and report there.
